logger/handlers/ringbuffer.go

```diff
@@ -0,0 +1,134 @@
+package handlers
+
+import (
+	"container/ring"
+	"fmt"
+	"log"
+	"net/http"
+	"regexp"
+	"strconv"
+
+	"github.com/deis/deis/logger/drain"
+	"github.com/deis/deis/logger/syslog"
+)
+
+var projectBuffer = make(map[string]*ring.Ring)
+var ringBufferSize int
+
+func addToBuffer(name string, message string) {
+	currentRing, ok := projectBuffer[name]
+	if !ok {
+		r := ring.New(ringBufferSize)
+		r.Value = message
+		projectBuffer[name] = r
+	} else {
+		r := currentRing.Next()
+		r.Value = message
+		projectBuffer[name] = r
+	}
+}
+
+func RingBufferHandler(bufferSize int, webServicePort int) *Handler {
+	ringBufferSize = bufferSize
+	h := Handler{
+		BaseHandler: syslog.NewBaseHandler(5, filter, false),
+	}
+	go h.ringBufferLoop()
```

Comment:
I think cancel of ringBufferLoop needed for example in case of changing options or changing working mode of working w/o restarts. But i don't known any priority user scenario for this.
In case of cephless cluster user(admin) may want to change size of buffer in real time, but i think importance of this scenario is low and implementation of this functionality will have more complexity, then i would like in this PR.